### PR TITLE
feat(siren): add list notifications api

### PR DESCRIPTION
**Pull request by manishdangi98**
Add List Notification API in Siren. 
This API list notifications based on the filter passed.
Supported filters are Type, Template, ReceiverSelectors and Labels.

### CODE REVIEW

**Comment by mabdh** [comment]
@manishdangi98 please add the description

File: gotocompany/siren/v1beta1/siren.proto [hunk 1]
```diff
@@ -973,4 +984,37 @@ message NotificationMessage {
...
+}
+
+message ListNotificationsRequest{
+  string type = 1;
+  string template = 2;
+  map<string, string> labels = 3 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    description: "query result based on notification labels. the label is written as map. eg, \"labels[key1]=value1\""
+  }];
+  repeated ReceiverSelectors receiver_selectors = 4 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    description: "query result based on receiver  selectors. the label is written as map. eg, \"[]receivers_selectors[key1]=value1\""
```
Comment (mabdh) [inline gotocompany/siren/v1beta1/siren.proto:996]:
```suggestion
    description: "query result based on receiver  selectors. the label is written as map. eg, \"[]receiver_selectors[key1]=value1\""
```

Is this how we use the selectors in query params?

File: gotocompany/siren/v1beta1/siren.proto [hunk 2]
```diff
@@ -973,4 +984,37 @@ message NotificationMessage {
...
+  }];
+}
+
+  message ReceiverSelectors {
+    map<string, string> receiver_selector  = 1;
+  }
```
Comment (mabdh) [inline gotocompany/siren/v1beta1/siren.proto:1002]:
```suggestion
message ReceiverSelectors {
  map<string, string> receiver_selector  = 1;
}
```

File: gotocompany/siren/v1beta1/siren.proto [hunk 3]
```diff
@@ -973,4 +984,36 @@ message NotificationMessage {
...
+}
+
+message ListNotificationsRequest{
+  string type = 1;
+  string template = 2;
+  map<string, string> labels = 3 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    description: "query result based on notification labels. the label is written as map. eg, \"labels[key1]=value1\""
+  }];
+  map<string, string> receiver_selectors = 4 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    description: "query result based on receiver  selectors. the label is written as map. eg, \"receivers_selectors[key1]=value1\""
```
[the remark this answers is not in the document]
Comment (manishdangi98) [reply gotocompany/siren/v1beta1/siren.proto]: Sorry I missed this, fixed now.

File: gotocompany/siren/v1beta1/siren.proto [hunk 4]
```diff
@@ -973,4 +984,36 @@ message NotificationMessage {
...
+    description: "query result based on notification labels. the label is written as map. eg, \"labels[key1]=value1\""
+  }];
+  map<string, string> receiver_selectors = 4 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    description: "query result based on receiver  selectors. the label is written as map. eg, \"receiver_selectors[key1]=value1\""
+  }];
+}
+
+message ReceiverSelectors {
+  map<string, string> receiver_selector  = 1;
+}
```
[the remark this answers is not in the document]
Comment (manishdangi98) [reply gotocompany/siren/v1beta1/siren.proto:1002]: While querying notifications from database getting for receiver_slectors as []maps. 
For example : 
```sql 
SELECT receiver_selectors
FROM notifications 
WHERE receiver_selectors @> '[{"type": "email"}]'::jsonb;
\\ Result 
[{"type": "slack_channel", "severity": "WARNING"}, {"type": "email", "severity": "WARNING"}]
```
Repeated is not supported in proto directly  `repeated map<string,string>`. 

For the response message this is added to return response of receiver_selectors.

[the remark this answers is not in the document]
Comment (mabdh) [reply gotocompany/siren/v1beta1/siren.proto:1002]: Oh right, this one is being used for the response right?
I mistakenly understood this for query param

**Review by mabdh — APPROVED** [review]

**Pull request merged**